src/plugin/diskusage_impl.cpp

```diff
@@ -111,15 +107,47 @@ quint64 DiskUsageWorker::calculateRpmSize(const QString &glob)
 
 quint64 DiskUsageWorker::calculateApkdSize(const QString &rest)
 {
-    Q_UNUSED(rest)
-    QDBusMessage msg = QDBusMessage::createMethodCall("com.jolla.apkd",
-            "/com/jolla/apkd", "com.jolla.apkd", "getAndroidAppDataUsage");
-
-    QDBusReply<qulonglong> reply = QDBusConnection::systemBus().call(msg);
-    if (reply.isValid()) {
-        return quint64(reply.value());
+    if (!m_apkd_data_queried) {
+        QDBusInterface apkd("com.jolla.apkd",
+                                "/com/jolla/apkd", "com.jolla.apkd", QDBusConnection::sessionBus());
+        QDBusReply<QDBusArgument> reply = apkd.call("getAndroidAppDataUsage");
+        if (reply.isValid()) {
+            int apkd_cache = 0;
+            const QDBusArgument output = reply.value();
+            output.beginStructure();
+            output >> m_apkd_app;
+            output >> m_apkd_data;
+            output >> apkd_cache;
+            m_apkd_data += apkd_cache;
+            output.endStructure();
+            m_apkd_run = calculateSize("/opt/alien");
+        } else {
+            qWarning() << "Android app data usage query failed. Trying Alien4 API";
```

Comment:
I'm just wondering a bit if this shouldn't be a warning. It's more like info level message. (Also it would be nice to have a logging category.)

Reply:
With a category there could be some debug printing going on for all of these size calculations, which I think should really be there. I'll do that.